Add a backward compatible function of mpv playlist-play-index

firmart opened the pull request:
The `mpv` command [playlist-play-index](https://mpv.io/manual/master/#command-interface-playlist-play-index) was only introduced recently in version 0.33. It would be useful to have a backward compatible wrapper for people haven't upgraded their `mpv` build.

kljohann commented:
Thanks, I'll have a look. In general I'd like to keep this library slim, so I wouldn't want to add too many extra functions. But if `playlist-play-index` is important for your workflow (or some other package) we can add it (also considering that 0.33 is [not available yet](https://repology.org/project/mpv/versions) for several widely used distributions).

firmart commented:
Thank you. Yes, it would be useful/fundamental for some package like [vuiet](https://github.com/mihaiolteanu/vuiet) to be able to jump around in a playlist. This PR comes from a [misunderstanding](https://github.com/mihaiolteanu/vuiet/issues/48) that any command in the mpv documentation would work with `mpv-run-command`. However, `(mpv-run-command "playlist-play-index" 0)` simply doesn't work for `mpv` version prior to `0.33`.

kljohann commented on mpv.el:
```diff
@@ -373,5 +373,37 @@ the echo area."
+	((and (pred numberp) (pred (= cur))) (mpv-seek 0))
+	("none" (mpv-set-property "pause" "no"))
+	((and (pred numberp) (pred (<= 0)))
+	 (cl-loop repeat (abs (- pos cur))
```
According to the linked documentation (https://mpv.io/manual/master/#command-interface-playlist-pos), `cur` might be `-1` if no entry is "current". This is not handled at the moment, right? Also:
> Before mpv 0.33.0, instead of returning -1, this property was unavailable if no playlist entry was current.

My version of mpv is too new to test this, but my assumption is that `mpv-get-property` would fail with `‘get_property’ failed: property not found` in that case?

kljohann replied:
Also, the mpv documentation indicates that `playlist-pos` is writable? So it _might_ be possible to replace the `cl-loop` with `(mpv-set-property "playlist-pos" pos)` in the `(!= pred cur)` case. Can you give it a try on your version of mpv?

kljohann reviewed (commented):
Looks nice, some comments inline.

firmart commented:
I fixed the code to reflect your suggestions. I wasn't able to test the case whenever `playlist-pos` returns `-1`, because I don't know exactly how to remove the `current` flag. Maybe I simply can't with my `mpv version v0.32`. I tried both `(mpv-set-property "playlist-pos" -1)`  and `(mpv-set-property "playlist-current-pos" -1)`. The former raised the error `unsupported format for accessing format` and the latter raised `property not found`.

As you said, `(mpv-set-property "playlist-pos" pos)` does work, though the documentation says
```
Writing the current value back to the property is subject to change. 
Currently, it will restart playback of the playlist entry. 
But in the future, writing the current value will be ignored. 
Use the playlist-play-index command to get guaranteed behavior.
```
we are fine as `mpv version 0.33` won't go through this path. So I replaced the `cl-loop`.

kljohann commented:
Looks good! On 0.33 I can get `playlist-pos = -1` by starting mpv with `--idle=yes` or `--idle=once`. It will be `-1` initially if no filename has been provided, and, in the case of `--idle=yes`, also after playback has finished:
```elisp
(mpv-start "--idle=once")
(mpv-get-property "playlist-pos") ;; returns `-1` on 0.33

(mpv-start "--idle=yes" "https://www.youtube.com/watch?v=ylXk1LBvIqU")
;; ... wait until playlist has finished playing ...
(mpv-get-property "playlist-pos") ;; returns `-1` on 0.33
```
Can you give this a try on 0.32?

kljohann commented on mpv.el:
```diff
@@ -395,14 +395,9 @@ https://mpv.io/manual/master/#command-interface-playlist-play-index"
 	("current" (mpv-seek 0))
 	((and (pred numberp) (pred (= cur))) (mpv-seek 0))
-	("none" (mpv-set-property "pause" "no"))
+	("none" (mpv-set-property "pause" "yes"))
```
In principle we should even enter idle mode here (according to the 0.33 man page), but I'm not sure how to do that on earlier versions. In 0.33 `(mpv-set-property "playlist-pos" -1)` would have that effect, but I doubt that this will work on earlier versions. So I think your current implementation is probably as close as we can get.
